src/utils/validations.js

```diff
@@ -231,3 +231,42 @@ export const validateBmcURL = value => {
 
   return validateURL(value);
 };
+
+const MAC_REGEX_COLON_DELIMITER =
+  '^(([0-9A-Fa-f]{2}[:]){19}[0-9A-Fa-f]{2})$|' + // 01:23:45:67:89:ab:cd:ef:00:00:01:23:45:67:89:ab:cd:ef:00:00
+  '^(([0-9A-Fa-f]{2}[:]){7}[0-9A-Fa-f]{2})$|' + // 01:23:45:67:89:ab:cd:ef
+  '^(([0-9A-Fa-f]{2}[:]){5}[0-9A-Fa-f]{2})$'; // 01:23:45:67:89:ab
+
+const MAC_REGEX_DASH_DELIMITER =
+  '^(([0-9A-Fa-f]{2}[-]){19}[0-9A-Fa-f]{2})$|' + // 01-23-45-67-89-ab-cd-ef-00-00-01-23-45-67-89-ab-cd-ef-00-00
+  '^(([0-9A-Fa-f]{2}[-]){7}[0-9A-Fa-f]{2})$|' + // 01-23-45-67-89-ab-cd-ef
+  '^(([0-9A-Fa-f]{2}[-]){5}[0-9A-Fa-f]{2})$'; // 01-23-45-67-89-ab
+
+const MAC_REGEX_PERIOD_DELIMITER =
+  '^(([0-9A-Fa-f]{4}.){9}[0-9A-Fa-f]{4})$|' + // 0123.4567.89ab.cdef.0000.0123.4567.89ab.cdef.0000
+  '^(([0-9A-Fa-f]{4}.){3}[0-9A-Fa-f]{4})$|' + // 0123.4567.89ab.cdef
+  '^(([0-9A-Fa-f]{4}.){2}[0-9A-Fa-f]{4})$'; // 0123.4567.89ab
+
+const COLON_DELIMITER = ':';
+const DASH_DELIMITER = '-';
+const PERIOD_DELIMITER = '.';
+
+// Validates that the provided MAC address meets one of following formats supported by the golang ParseMAC function:
+// IEEE 802 MAC-48, EUI-48, EUI-64, or a 20-octet IP over InfiniBand link-layer address
+// https://golang.org/pkg/net/#ParseMAC
+export const isValidMAC = mac => {
+  if (mac.length < 14) {
+    return false;
+  }
+
+  let regex;
+  if (mac[2] === COLON_DELIMITER) {
+    regex = new RegExp(MAC_REGEX_COLON_DELIMITER);
```

Comment:
it is not necessary to call `RegExp` each time. I would rather create it when creating the const variable.

Reply:
Done.

Reply:
Done.